File: src/main/java/camera/UriUtil.java
package camera;

import java.io.File;
import java.net.URI;
import java.net.URISyntaxException;

class UriUtil
{
    private UriUtil() { }

    static File uriToFile(String uriString) {
        if (uriString == null) {
            return null;
        }
        final URI fileURI;
        try {
            fileURI = new URI(uriString);
        } catch (URISyntaxException e) {
            return null;
        }
        final File file;
        try {
            file = new File(fileURI);
        } catch (IllegalArgumentException e) {
            return null;
        }
        return file;
    }
}

File: src/main/java/camera/UriUtil.kt
package camera

import java.io.File
import java.net.URI
import java.net.URISyntaxException

fun uriToFile(uriString: String?): File? {
    if (uriString == null) {
        return null
    }
    val fileURI = try {
        URI(uriString)
    } catch (e: URISyntaxException) {
        return null
    }
    return try {
        File(fileURI)
    } catch (e: IllegalArgumentException) {
        return null
    }
}

